### PR TITLE
Fixes Scale Shot not activating + Known Failing Round test

### LINKED ISSUE #5148: Round preventing Pokemon to make a Move
### Description

If the AI manages to setup a double Round, it can change the Turnorder dynamically. This seems to create some issues and can make one of the Players Pokemon unable to move. Here is a clip for reference where Leavanny was not allowed to get its move off. (It's not because of Throat Spray, I tested further with this)
Note: The Partner switching out is necessary to recreate this. For some reason it has to be the Pokemon in the right switching out to trigger this Bug.

https://github.com/user-attachments/assets/aeb9bb9b-528e-43ec-b849-f9615755e751

### Version

1.9.0 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

._.terra._.

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #5148

Also fixes Scale Shot not activating when `gMultiHitCounter` was not decreased to zero while opposing mon fainted.

### CODE REVIEW

File: test/battle/move_effect/multi_hit.c [hunk 1]
```diff
@@ -183,3 +183,51 @@ SINGLE_BATTLE_TEST("Endure does not prevent multiple hits and stat changes occur
...
+        PLAYER(SPECIES_WOBBUFFET) { Item(ITEM_LOADED_DICE); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_SCALE_SHOT); }
+    } SCENE {
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_SCALE_SHOT, player);
+        MESSAGE("Hit 5 time(s)!");
```
Comment (AsparagusEduardo) [inline test/battle/move_effect/multi_hit.c:201]: If it's hitting 5 times, why is it only playing the animation 4 times?

**Review by AsparagusEduardo — COMMENTED** [review]
Small question, everything else lgtm :)

File: test/battle/move_effect/multi_hit.c (hunk 1 above)
Comment (AlexOn1ine) [reply test/battle/move_effect/multi_hit.c:201]: Good question. It should be 4 times. It passes because it can randomly pass with 5 hits.

Comment (AlexOn1ine) [inline test/battle/move_effect/multi_hit.c:201]:
```suggestion
        MESSAGE("Hit 4 time(s)!");
```

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**